lib/topics_filter.rb

```diff
@@ -48,14 +53,13 @@ def filter_from_query_string(query_string)
             filter_categories(
               category_slugs: category_slugs.split(delimiter),
               exclude_subcategories: key_prefix.presence,
-              or_clause: category_or_clause,
             )
-
-          category_or_clause = true
         end
       end
     end
 
+    @registered_scopes.each { |_key, value| @scope = value[:block].call(*value[:params].values) }
```

Comment:
Introduced this pattern so that we can apply a scope only after we've scanned through all the filters and collected what we need.